src/virtual_dom/vnode.rs

```diff
@@ -31,6 +31,18 @@ pub enum VNode {
     VRef(Node),
 }
 
+impl VNode {
+    pub fn key(&self) -> Option<String> {
+        match self {
+            VNode::VTag(vtag) => vtag.key.clone(),
```

Comment:
this `clone()` is too eager. It would be better if `VNode::key()` returned a reference